### PR TITLE
Add adoc version of original 10 Steps SOE document

### PR DESCRIPTION
**Pull request by wzzrd**
As discussed, a PR to merge the original version of the 10 steps into the documentation repo in order to facilitate working on this with Red Hatters, customers and Foreman community members alike.

### CODE REVIEW

**Comment by melcorr** [comment]
cc @ehelms

File: guides/doc-10-steps/main.adoc
Comment (lzap) [inline guides/doc-10-steps/main.adoc:5]: Ehm. Hmmm.

File: guides/doc-10-steps/main.adoc
Comment (lzap) [inline guides/doc-10-steps/main.adoc:25]: Leftover?

File: guides/doc-10-steps/Readme.md
Comment (lzap) [inline guides/doc-10-steps/Readme.md:2]: I do not have the context, but one idea that hits me is what if this grows up to 11 steps? Probably not the best title, maybe too not formal?

**Comment by melcorr** [comment]
@lzap this is the original content of the SOE doc that Satellite used to have for customers. 
The idea is to take the base files and start to rework them over time with volunteers. 
Making it available for the community.

**Comment by maximiliankolb** [comment]
hey everyone. I had a quick look at the PR. It's quite large tbh.

- `Red Hat solutions provide additional benefits, such as improved Total Cost of Ownership (TCO), better IT productivity, and increased agility, to drive your business forward. Red Hat Enterprise Linux with Red Hat Satellite provides an ideal platform for standardized operating environments with lower TCO and greater IT efficiency and productivity, adding to the bottom line and increasing business agility and competitiveness.` < I am unsure if this should really be part of the Foreman documentation. It reads like marketing material.
- The PR as is does not adhere to the "one sentence per line" rule yet. > This makes it hard to review commits on top of your feature branch if I were to suggest/make changes here and there.
- The PR as is does not make use of any asciidoc attributes yet.
- Has this been tested with "pure" upstream software, i.e. foreman etc?
- `This is a Customer-facing document between Red Hat, Inc. and {customer} (“Client”).` ?
- `guides/doc-10-steps/document/0000-preface.adoc` < what license is this under?
- `end-to-end example of software life-cycle management using Red Hat Satellite 6` < this sounds really promising IMHO!
- `“why”` < maybe change highlighting to something more asciidoc friendly, like `_` or `*`
- `Some tools used and items documented inside are not covered by Red Hat’s Global Support Services support contracts and coverage ...` < if it's part of the community docs, then this should be at least in conditionals.
- `customers` < should this say `users`? Foreman does not really has customers but rather users. Maybe we can use an attribute with "upstream/foreman > users" and "downstream > customers"?
- `by emailing refarch-feedback@redhat.com` > maybe we can reword this to `Please open a PR on github.com/theforeman/foreman-documentation or submit your feedback to our community forum at community.theforeman.org.`
- `https://twitter.com/RedHatRefArch` does not exist (anymore?); I did not (want to) check facebook and google+.
- `Recommended Red Hat Training Offerings` < nice section, maybe we can also adjust this to "upstream > foreman docs and foreman media on theforeman.org" and "downstream > Company X trainings".
- `Recommended Red Hat Consulting Offerings` same as above
- `https://access.redhat.com/documentation/en-US/Red_Hat_Satellite/6.1/html/Installation_Guide/chap-Red_Hat_Satellite-Installation_Guide-Installing_Red_Hat_Satellite_Server.html[https://access.redhat.com/documentation/en-US/Red_Hat_Satellite/6.1/html/Installation_Guide/chap-Red_Hat_Satellite-Installation_Guide-Installing_Red_Hat_Satellite_Server.html]` < title needs to be fixed and also the version. See #545 where I propose using an attribute such as "AccessRedHatDotCom" to specify the version of the Satellite docs.
- `https://access.redhat.com/articles/1380493[https://access.redhat.com/articles/1380493[ ]]https://access.redhat.com/articles/1380493[https://access.redhat.com/articles/1380493]` < broken
- `1. _Infrastructure_ ➤ _Compute Resources_ ➤ _New Compute Resource_ +` < replace arrows with `>`
- `capsule-installer` < use an attribute to differentiate between upstream and downstream
- `. Click ‘Content’ -> ‘Content Views’.` < "wrong" quotes; we use `*` for highlighting in this case.
- `*8. Retrieve the Kickstart Profile*` < use dots to create ordered lists
- `==== 2. Update the Affected Composite Content Views` < use `:numbered:` and `:numbered!:` instead


I was skimming most of it. I would prefer to have much smaller PRs but I would also contribute time to get this up to Foreman documentation conventions etc. This is not a review, just some stuff I noticed while reading.

Overall, it could be a great addition to the Foreman documentation 👍

**Comment by melcorr** [comment]
@lzap @maximiliankolb I think the idea is that we would host the source here. 
almost all of the source needs to be reworked. Some might be deleted completely. 
Nothing is fit for publication. There are a team of folks volunteering to rebuild this. 
So, do we accept this as the base to work on, not published anywhere, or should we pick it apart in this PR?

**Comment by maximiliankolb** [comment]
I did not mean to pick the PR apart, but I actually believe it's too big to review it in one go. How would you like to proceed?

**Comment by melcorr** [comment]
> I did not mean to pick the PR apart, but I actually believe it's too big to review it in one go. How would you like to proceed?

I think the original intent was just that the source would be stored here, then worked on later. 
I would be happy with that. It won't be published anywhere, but perhaps we need to stop it from failing the build. 
I'll get to this next week.

**Comment by maximiliankolb** [comment]
I am somewhat uncomfortable merging this to master.

Moving forward, I propose a **temporary** "devel" branch to work on this together, so we can work on it collectively until it's ready. Note: nobody should be able to force-push into this branch; we could clean up its history once we merge this to master.

**Comment by melcorr** [comment]
@lzap what do you think of @maximiliankolb's proposal of a devel branch for this?

**Comment by lzap** [comment]
I am not against temporary branches, it is also possible to allow other people with push access to write to your local branches too. So it does not have to be a branch in the upstream repo. Up to you.

**Comment by maximiliankolb** [comment]
I am totally fine with the temporary branch on any repo. My main idea was to have a temporary, yet "stable" branch without force pushing, so that we can work on it (push directly or create PR to it; both is fine for me) as long as it takes, and then squash and merge to foreman docs. Again, I would definitely contribute to preparing the content.

**Comment by ehelms** [comment]
My view has been that master is the devel branch. And what we put into the code repository, does not necessarily have to be published to the docs right away while we work on it. If we create multiple branches or collaborate on someone else's remote we run the risk of making collaboration harder. To summarize what I am saying, I think we should support and differentiate developing documentation and publishing it. And for me, the best place to collaboratively develop is in the master branch of this repository.

I'm interesting in this topic for this work but also because I wish to start building up and writing some architecture related documents with folks. However, we'd like to start writing the pieces first, in this repository, and then figure out the structure and collaboration down the road. We'd like folks to see and collaborate on what we are building up and for me that would mean the master branch of this repository. And putting the docs in some new folder to start with (e.g. architecture_docs/).

**Comment by maximiliankolb** [comment]
I change my mind because I really like the "make (contributing to) documentation more inclusive" argument @ehelms . 👍 

Do we have a process in place to go from "master" (I would love to see this branch being renamed to "devel" or "main") to stable release branches? AFAIK not yet.

**Comment by ehelms** [comment]
> I would love to see this branch being renamed to "devel" or "main"

Would be good for an issue so we can discuss and then take action. I like matching Foreman's develop and I think it does make it more obvious that its the in progress place for developing documentation.

> Do we have a process in place to go from "master" (I would love to see this branch being renamed to "devel" or "main") to stable release branches?

Solid question, branching is sorta "all in". That's part of why I focused more on published vs. unpublished. We do not have a clear status for either on how it goes from one to the other. I suppose the simple answer is a developer opens a PR to start publishing when they think something is ready. Would a draft vs GA status for published content be worth while?

**Comment by ehelms** [comment]
An idea from @ekohl  is to introduce a `drafts/` folder to identify documentation in progress and within that we could then have sub-folders. So this change would go to `drafts/10-steps/` and architecture documents could go in `drafts/architecture`.

**Comment by calvingsmith** [comment]
Hey has there been any decision as to where to put this content?

**Comment by melcorr** [comment]
Hi @calvingsmith sorry for the delay, I'm waiting on agreement on #625, which is blocking this PR.

**Comment by maximiliankolb** [comment]
https://github.com/theforeman/foreman-documentation/pull/625 has been merged. Please rebase your PR and put all files into the `drafts/` directory. Then we can merge this PR and work/improve on it together. @wzzrd @calvingsmith

**Comment by wzzrd** [comment]
Rebased and force pushed. Hope my git foo doesn't end life as we know it :)

File: drafts/doc-10-steps/document/0000-preface.adoc
Comment (maximiliankolb) [inline drafts/doc-10-steps/document/0000-preface.adoc:6]: Would it make sense to adjust the [license](https://github.com/theforeman/foreman-documentation/blob/master/LICENSE) before merging to master?

**Review by maximiliankolb — COMMENTED** [review]
I made one comment regarding the license. Another potential issue I see are the images. How do you want to handle this? Branding is ok IMHO until we create a upstream friendly alternative. However, I would appreciate it if there was a clear license for all content, text and images.

Apart from that; i did not read everything, but it looks good for a first draft.

File: drafts/doc-10-steps/document/0000-preface.adoc
Comment (wzzrd) [reply drafts/doc-10-steps/document/0000-preface.adoc:6]: We should probably remove that file (0000-preface.adoc) completely, assume the same CC license as the main repo. 

Regarding the images, we'll want to eventually replace those images completely with something that has community branding to begin with. Maybe remove them too for now?

Comment (maximiliankolb) [reply drafts/doc-10-steps/document/0000-preface.adoc]: Good. I will start the GHA again; once this is green, I will approve this PR.

**Review by maximiliankolb — APPROVED** [review]
LGTM

**Comment by maximiliankolb** [comment]
> LGTM

If you agree, please press the button^^ cc @melcorr @lzap

**Comment by melcorr** [comment]
Finally!

**Comment by calvingsmith** [comment]
Yay!

**Pull request merged**
